src/ar_gateway_middleware.erl

```diff
@@ -237,11 +253,122 @@ get_tx_from_hash_1(TXID) ->
 		Filename -> {ok, TXID, Filename}
 	end.
 
-serve_tx(Filename, Req) ->
+serve_tx(Filename, Path, Fallback, Req, Env) ->
 	TX = ar_storage:read_tx_file(Filename),
-	ContentType = proplists:get_value(<<"Content-Type">>, TX#tx.tags, "text/html"),
-	Headers = #{<<"content-type">> => ContentType},
+	ContentType =
+		case lists:keyfind(<<"Content-Type">>, 1, TX#tx.tags) of
+			{<<"Content-Type">>, V} -> V;
+			false -> <<"text/html">>
+		end,
+	case {ContentType, Path} of
+		{?MANIFEST_CONTENT_TYPE, <<>>} ->
+			serve_manifest(TX, Req);
+		{?MANIFEST_CONTENT_TYPE, _} ->
+			serve_manifest_path(TX, Path, Fallback, Req, Env);
+		{_, <<>>} ->
+			serve_plain_tx(TX, ContentType, Req);
+		_ ->
+			bad_request(Req)
```

Comment:
Shouldn't we also consider the fallback in this case? Specifically, a custom domain request to a custom path, which we pass on to other middleware, may end up here, right?

Reply:
Good catch! Addressed in d118d06e